crossplane/objects.py

```diff
@@ -33,23 +41,77 @@ def get(self, directive):
         """
         return []
 
-    def to_dict(self):
+    def to_crossplane(self, include_meta=True):
         result = {}
 
+        # exclude slots unique to object layer
         for slot in self.__slots__:
-            if slot not in ('parent',):
+            if slot not in self.__meta_slots__:
                 result[slot] = getattr(self, slot)
 
         return result
 
+    def hexdigest(self):
+        crossplane_dict = self.to_crossplane(include_meta=False)
+
+        if PYTHON3:
+            return hashlib.sha256(
+                str(crossplane_dict).encode('utf-8')
+            ).hexdigest()
+        else:
+            return hashlib.sha256(
+                unicode(crossplane_dict).encode('utf-8')
+            ).hexdigest()
+
+
+class NginxDirective(CrossplaneObject):
+    __slots__ = ('parent', 'directive', 'line', 'args', 'includes')
+    __meta_slots__ = ('parent',)
```

Comment:
what's a "meta slot"? That's pretty vague. Just add an underscore to the "meta variables" like `'_parent'` and `'_index'` and then in `get()` or whatever exclude variables that start with `_`. It'd be like two birds with one stone or whatever because you could get rid of the whole "meta slots have different behaviors" thing and also it'd make it clear which variables were inside the original dict and which weren't.